linkml/generators/common/ifabsent_processor.py

```diff
@@ -61,6 +65,9 @@ def _map_to_default_value(
         if mapped:
             return custom_default_value
 
+        if ifabsent_default_value == self.DEFAULT_RANGE_SPECIAL_CASE:
```

Comment:
I'm wondering if it would be better to also have an `if` statement for the cases managed by `_map_uri_special_case(..)` and `_map_curie_special_case(..)` ? 🤔 

What I mean is something like this:

```python
        if slot.range == Uri.type_name:
          if default_value in URI_SPECIAL_CASES:
            return self.map_uri_special_case(ifabsent_default_value, slot, cls)
          return self.map_uri_default_value(ifabsent_default_value, slot, cls)

        if slot.range == Curie.type_name:
          if default_value in CURIE_SPECIAL_CASES:
            return self.map_curie_special_case(ifabsent_default_value, slot, cls)
          return self.map_curie_default_value(ifabsent_default_value, slot, cls)

        if slot.range == Uriorcurie.type_name:
            if default_value in URI_SPECIAL_CASES:
              return self.map_uri_special_case(ifabsent_default_value, slot, cls)
            if default_value in CURIE_SPECIAL_CASES:
              return self.map_curie_special_case(ifabsent_default_value, slot, cls)            
            return self.map_uri_or_curie_default_value(ifabsent_default_value, slot, cls)
```

What do you think ?

Reply:
(I forgot to publish this message last week, it was lying there in my browser 😬 )

Reply:
fair proposal - the idea here was that we want to limit the switching logic that happens in this method as much as possible. basically this is already so stacked up that if any subclass needed to override it the entire purpose of the class tree would be lost. there is already in these two cases of the python and shacl handlers need to handle these switches in different ways, so i am deferring that responsibility to the sub-generators and providing these convenience methods to them if they need it.

Reply:
oop forgot to add - the reason that the `DEFAULT_RANGE_SPECIAL_CASE` gets added here (and is explicitly labeled as a special case to prominently) is that it's a type-less parameter, where the others are based on detection of the type of the range. it's used exactly once in the metamodel, and as far as i can tell that's the only time you would use the value of `default_range` as the `ifabsent` value.

Reply:
as Gödel demands, no self-describing schema language can exist without a few axioms taken for granted (aka special cases)

Reply:
Ok I understand your logic 👍 
That's good to me so 😊